pandas/core/arrays/interval.py

```diff
@@ -223,12 +229,14 @@ def _simple_new(
         if dtype is not None:
             # GH 19262: dtype must be an IntervalDtype to override inferred
             dtype = pandas_dtype(dtype)
-            if not is_interval_dtype(dtype):
+            if is_interval_dtype(dtype):
+                dtype = cast(IntervalDtype, dtype)
+                if dtype.subtype is not None:
+                    left = left.astype(dtype.subtype)
+                    right = right.astype(dtype.subtype)
+            else:
```

Comment:
why change the logic here? If the rest of the PR just adds type hints, maybe it would be safer to keep refactorings to a separate PR?

Reply:
I changed the logic because otherwise the Mypy checks will not pass.

Reply:
this is fine (you just moved and changed the logical test), but puzzled why we don't need the clause on L229

Reply:
cc @jbrockmendel if any idea here